gendox-frontend/src/context/AuthContext.js

```diff
@@ -24,159 +16,199 @@ const defaultProvider = {
-        setLoading(false)
-      }
+    try {
+      const response = await axios.get(authConfig.getProfile, {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${storedToken}`,
+        },
+      });
+
+      // Add 'role': 'admin' to the userDataResponse.data object
+      const userData = { ...response.data, role: "admin" };
+      setUser(userData);
+      dispatch(userDataActions.getUserData(userData));
+      setLoading(false);
+    } catch (error) {
+      console.error("Error during auth initialization:", error);
+      handleLogout();
+    } finally {
+      setLoading(false);
     }
+  };
 
-    initAuth()
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  useEffect(() => {
+    initAuth();
+  }, []);
 
-  // Handling Active Organization and Project based on URL changes
   useEffect(() => {
-    const { organizationId, projectId } = router.query
+    const { organizationId, projectId } = router.query;
+    const storedToken = window.localStorage.getItem(authConfig.storageTokenKeyName)
+
     if (user && user.organizations) {
-      const updatedActiveOrganization = user.organizations.find(org => org.id === organizationId)
+      const updatedActiveOrganization = user.organizations.find(
+        (org) => org.id === organizationId
+      );
       if (updatedActiveOrganization) {
-        setActiveOrganization(updatedActiveOrganization)
-        dispatch(activeOrganizationActions.getActiveOrganization(updatedActiveOrganization))
-        window.localStorage.setItem(authConfig.selectedOrganizationId, updatedActiveOrganization.id)
-
-        const updatedActiveProject = updatedActiveOrganization.projects.find(proj => proj.id === projectId)
+        
+        dispatch(
+          fetchOrganizationById({
+            organizationId: updatedActiveOrganization.id,
+            storedToken          
+          })
+        );
+        window.localStorage.setItem(
+          authConfig.selectedOrganizationId,
+          updatedActiveOrganization.id
+        );
+        const updatedActiveProject = updatedActiveOrganization.projects.find(
+          (proj) => proj.id === projectId
+        );
         if (updatedActiveProject) {
-          setActiveProject(updatedActiveProject)
-          dispatch(activeProjectActions.getActiveProject(updatedActiveProject))
-          window.localStorage.setItem(authConfig.selectedProjectId, updatedActiveProject.id)
+          dispatch(
+            fetchProjectById({
+              organizationId: updatedActiveOrganization.id,
+              projectId: updatedActiveProject.id,
+              storedToken            
+            })
+          );
+          window.localStorage.setItem(
+            authConfig.selectedProjectId,
+            updatedActiveProject.id
+          );
         }
       }
     }
-  }, [user, router.query.organizationId, router.query.projectId])
+  }, [user, router.query.organizationId, router.query.projectId]);
```

Comment:
I don't think that this is the right place to handle URL changes. 
Let's discuss it offline and create a ticket for later